### ISSUE TITLE
[TypeScriptToTypeBox] Non-consistent work with non-existing types

### ISSUE THREAD

**Issue by ilyhalight** [sinclairzx81/typebox-codegen#40]
Hi. I noticed that if a type that doesn't exist is involved in the generation of one of the properties, then typebox-codegen will save it. At the same time, if we create a new interface extending another non-existent type, then typebox-codegen will remove it in some cases

**Context**: in this case, typebox-codegen is used to generate typebox types from multiple typescript files:
- jmap.ts
- mail.ts
- client.ts

The mail.ts file contains imports of all types from jmap.ts (`import * as JMAP from "./jmap";`). I read the contents of these files and turn them into typebox types through typebox-codegen. Since typebox-codegen can't import the file, the generator thinks that JMAP.* doesn't exist. (a similar reproduction option, but without using import from file, can be viewed at the end of the issue)

**[v1 extends (invalid)] An example of an interface extending another type that doesn't exist (JMAP.GetResponse doesn't exists, generated invalid Type.Composite):**
```ts
// typescript (JMAP.GetResponse doesn't exists)
export interface MailGetResponse extends JMAP.GetResponse {
  test: string;
}
```

```ts
//  typebox-codegen (Unhandled: PropertyAccessExpression JMAP.GetResponse)
import { Type, Static } from '@sinclair/typebox'


export type MailGetResponse = Static<typeof MailGetResponse>
export const MailGetResponse = Type.Composite([, Type.Object({ // <-- invalid Type.Composite
test: Type.String()
})])
```

```ts
//  typebox expected
import { Type, Static } from '@sinclair/typebox'


export type MailGetResponse = Static<typeof MailGetResponse>
export const MailGetResponse = Type.Composite([JMAP.GetResponse, Type.Object({
test: Type.String()
})])
```

**[v2 extends (valid)] An example of an interface extending another type that doesn't exist (TestResponse doesn't exists, all valid):**
The code is the same as in [v1], but instead of JMAP.GetResponse is used by TestResponse (also doesn't exist). typebox-codegen returns fully valid types
```ts
// typescript (TestResponse doesn't exists)
export interface MailGetResponse extends TestResponse {
  test: string;
}
```

```ts
// typebox-codegen (valid)
import { Type, Static } from '@sinclair/typebox'


export type MailGetResponse = Static<typeof MailGetResponse>
export const MailGetResponse = Type.Composite([TestResponse, Type.Object({
test: Type.String()
})])
```

```ts
//  typebox expected
import { Type, Static } from '@sinclair/typebox'


export type MailGetResponse = Static<typeof MailGetResponse>
export const MailGetResponse = Type.Composite([TestResponse, Type.Object({
test: Type.String()
})])
```

**[valid] An example of a type with a non-existent property that is also generated correctly (all valid):**
```ts
// typescript
export type Mailbox = {
  id: JMAP.Id; // <-- doesn't exist
  name: string;
}
```

```ts
//  typebox-codegen (valid)
import { Type, Static } from '@sinclair/typebox'


export type Mailbox = Static<typeof Mailbox>
export const Mailbox = Type.Object({
id: JMAP.Id,
name: Type.String()
})
```

```ts
//  typebox expected
import { Type, Static } from '@sinclair/typebox'


export type Mailbox = Static<typeof Mailbox>
export const Mailbox = Type.Object({
id: JMAP.Id,
name: Type.String()
})
```

It can be reproduced using the following code:
```ts
import * as Codegen from "@sinclair/typebox-codegen";

const code = Codegen.TypeScriptToTypeBox.Generate(`
export interface MailGetResponse extends JMAP.GetResponse {
  test: string;
}
`);

console.log(code);

```

**Comment by sinclairzx81** [comment]
@ilyhalight Hi,

I've pushed a quick fix for this on 0.9.15, such that the generator will now emit for non-existent types. Just be mindful that this will only work for the `TypeScriptToTypeBox` generator as the other generators require code `eval()` of generated TypeBox, and where evaluation will fail for missing JS values.

This update has also been applied to the Workbench (but you will need to specify the missing types there), I have however added support for namespaced types. You can demo things at the following URL

[Workbench Link Here](https://sinclairzx81.github.io/typebox-workbench/?share=Ly8gbmFtZXNwYWNlIHR5cGVzIHN1cHBvcnRlZCBvbiBjb2RlZ2VuIC8gMC45LjE1Cm5hbWVzcGFjZSBKTUFQIHsKICBleHBvcnQgaW50ZXJmYWNlIEdldFJlc3BvbnNlIHsKICAgIHN0YXR1czogbnVtYmVyCiAgfQp9CgpleHBvcnQgaW50ZXJmYWNlIE1haWxHZXRSZXNwb25zZSBleHRlbmRzIEpNQVAuR2V0UmVzcG9uc2UgewogIHRlc3Q6IHN0cmluZzsKfQ%3D%3D)

Let me know if this helps
Cheers
S

**Comment by ilyhalight** [comment]
I tested it in TypeScriptToTypeBox - it works, thanks again :)
